plugins/modules/panos_commit_firewall.py

```diff
@@ -0,0 +1,210 @@
+        default: True
+'''
+
+EXAMPLES = r'''
+- name: commit candidate configs on firewall
+  panos_commit_firewall:
+    provider: '{{ credentials }}'
+
+- name: commit changes by specified admins on firewall
+  panos_commit_firewall:
+    provider: '{{ credentials }}'
+    admins: ['netops','secops','cloudops']
+    description: 'Saturday change window'
+
+- name: commit only policy and object changes on firewall
+  panos_commit_firewall:
+    provider: '{{ credentials }}'
+    exclude_device_and_network: True
+'''
+
+RETURN = r'''
+jobid:
+  description: The ID of the PAN-OS commit job.
+  type: int
+  returned: always
+  sample: 49152
+starttime:
+  description: The time the commit job started.
+  type: str
+  returned: on success
+  sample: 2020, 9, 6, 17, 56, 20
+endtime:
+  description: The time the commit job ended.
+  type: str
+  returned: on success
+  sample: 2020, 9, 6, 17, 56, 35
+runtime:
+  description: The elapsed time (in seconds) of the commit job.
+  type: int
+  returned: on success
+  sample: 15
+details:
+  description: Commit job completion messages.
+  type: str
+  returned: on success
+  sample: Configuration committed successfully
+warning:
+  description: Commit job warnings.
+  type: list
+  returned: on success
+  sample: ["","","","","","","","","","",""]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.paloaltonetworks.panos.plugins.module_utils.panos import get_connection
+
+try:
+    from panos.firewall import FirewallCommit
+    from dateutils import parser
+except ImportError:
+    pass
+
+
+def main():
+    # Instantiate the connection helper
+    helper = get_connection(
+        min_pandevice_version=(1, 0, 0),
+        min_panos_version=(8, 0, 0),
+        argument_spec=dict(
+            description=dict(type='str'),
+            admins=dict(type='list'),
+            exclude_device_and_network=dict(type='bool'),
+            exclude_shared_objects=dict(type='bool'),
+            exclude_policy_and_objects=dict(type='bool'),
+            force=dict(type='bool'),
+            sync=dict(type='bool', default=True)
+        )
+    )
+
+    # Initialize the Ansible module
+    module = AnsibleModule(
+        argument_spec=helper.argument_spec,
+        supports_check_mode=False,
+        required_one_of=helper.required_one_of
+    )
+
+    # Verify libs are present, get the parent object.
+    parent = helper.get_pandevice_parent(module)
+
+    # Construct the commit command
+    cmd = FirewallCommit(
+        description=module.params['description'],
+        admins=module.params['admins'],
+        exclude_device_and_network=module.params['exclude_device_and_network'],
+        exclude_shared_objects=module.params['exclude_shared_objects'],
+        exclude_policy_and_objects=module.params['exclude_policy_and_objects'],
+        force=module.params['force']
+    )
+
+    # Execute the commit
+    # commit_results = dict(changed=False, jobid=0)
+    commit_results = {}
```

Comment:
Needs to be:

```python
commit_results = {'changed': False, 'jobid': 0}
```

Otherwise there's nothing in `commit_results` if no commit is needed.